### ISSUE TITLE
Kubewarden policies: show "signed" badge

### ISSUE THREAD

**Issue by flavio** [artifacthub/hub#2146]
**Is your feature request related to a problem? Please describe.**

Kubewarden policies are WebAssembly modules, basically binary blobs, that are distributed using container registries.
Kubernetes operators need a way to prove the trustworthiness of a policy. This is currently solved by using the Sigstore project.

Kubewarden policies can be signed and verified in the same way as container images.

**Describe the solution you'd like**

It would be great to have ArtifactHub show a "signed badge" (like the one shown [here](https://artifacthub.io/packages/container/goreleaser/goreleaser), next to the container image name) for all the policies that have been signed using Sigstore.

**Comment by tegioz** [comment]
Deployed!

<img width="1716" alt="Screen Shot 2022-07-26 at 07 23 20" src="https://user-images.githubusercontent.com/1213902/180929703-646a5a40-e0ff-4659-807e-6ce69016f2e1.png">

**Comment by flavio** [comment]
Thanks a lot, this is amazing! :heart:

**Comment by tegioz** [comment]
No worries! Please note that we are checking the signature on the image named `policy`. Not sure if it's possible for a Kubewarden policy to list multiple images, but for now it's important that it's named that way for the cosign check to work.

 Happy to change how this works though, we could verify that *all* images provided are signed to mark the package as signed, or do it some other way 🙂

**Comment by tegioz** [comment]
If there will always be just one image, we can also remove that name requirement.

**Comment by flavio** [comment]
> No worries! Please note that we are checking the signature on the image named `policy`. Not sure if it's possible for a Kubewarden policy to list multiple images, but for now it's important that it's named that way for the cosign check to work.
>
> Happy to change how this works though, we could verify that _all_ images provided are signed to mark the package as signed, or do it some other way slightly_smiling_face

I don't think it would make sense for a single kubewarden policy to have multiple images. I would even consider that an error :thinking: 

We can leave your check as it currently is (looking for the `policy` image). Otherwise, ArtifactHub could prevent the scenario of multiple images images being mentioned inside of the `artifacthub-pkg.yml` or a Kubewarden policy; having done that there would be only 1 image to be verified. What do you think?

**Comment by tegioz** [comment]
One question @flavio: the case where a Kubewarden policy doesn't define an image in the AH metadata file, should we treat it as an error too?

At the moment multiple images are allowed and we are going to restrict that for Kubewarden policies as we discussed earlier. But I was wondering if we should also enforce **requiring** one to be present, as this is an optional field right now.

Please note that repositories with errors won't be indexed. Publishers will be notified by email when something goes wrong during the tracking process and they'll be able to inspect the errors from the control panel.

And one more thing if the response is affirmative 😅: if we end up requiring an image with a pre-defined named to be listed, would you like us to provide some generic installation instructions for Kubewarden policies? I've noticed that they are mostly the same except for the image url, so we could build that based on the image ourselves. This should make it easier for users to publish policies 🙂

**Comment by tegioz** [comment]
> I don't think it would make sense for a single kubewarden policy to have multiple images. I would even consider that an error 🤔

What do you think about the case where users may want to publish the same policy image to multiple registries and list all options on AH? Some users already do that for other artifacts, just in case a registry is down or because consumers may prefer to pull images from a specific one. Maybe it still makes sense to allow more than one image.

The offer about generic installation instructions still stands though 🙂 We could allow maybe two images and require a minimum of 1, one named `policy` and another named maybe `policy-alternative-location` or something similar, up to you. Installation instructions could display a select to switch between both installation sources when available. In this case, we'd be enforcing that up to 2 images can be present, and that they use the expected names. What do you think?

**Comment by flavio** [comment]
That's a good point. How do you handle Container images published on Artifact Hub that have multiple `image` entries in regards to cosign?

**Comment by tegioz** [comment]
Only two other kinds support cosign signature verification at the moment, Helm charts and Container Image kinds. Both are slightly different than the Kubewarden case in the sense that the repository URL used on AH is the actual image, it's not provided separately. So we mark the package as signed if the image in the referenced in the repository URL is signed. Other images provided via annotations or in the metadata file are not checked.

In the Kubewarden case, I'd suggest checking both, specially if we are going to provide installation instructions that allow users switching between the two installation sources, when available.

To summarize..

- [ ] Kubewarden policies packages *must* list at least one image in the AH metadata file, named `policy`
- [ ] A second image, named `policy-alternative-location`, *can* be provided optionally
- [ ] A Kubewarden policy package will be considered signed when *all* images provided are signed
- [ ] We'll provide default installation instructions based on the images defined above that will be displayed when no specific instructions are provided via the `install` field in the metadata file (so that you can override if needed)
- [ ] We need to add this information to the repositories guide docs 😉 

Does that make sense? 🙂

**Comment by flavio** [comment]
It makes a lot of sense. Go ahead with it :heavy_plus_sign: 

I especially love the "We'll provide default installation instructions" point, because maintaining this file in sync with the `images` is painful.
